src/openms/source/KERNEL/FeatureMap.cpp

```diff
@@ -193,12 +202,21 @@ namespace OpenMS
     unassigned_peptide_identifications_.insert(unassigned_peptide_identifications_.end(), rhs.unassigned_peptide_identifications_.begin(), rhs.unassigned_peptide_identifications_.end());
     data_processing_.insert(data_processing_.end(), rhs.data_processing_.begin(), rhs.data_processing_.end());
 
+    Size n_old_features = size();
     // append features:
     this->insert(this->end(), rhs.begin(), rhs.end());
 
     // todo: check for double entries
     // features, unassignedpeptides, proteins...
 
+    // merge IDs (new format):
```

Comment:
would it make sense to have a mergeAndUpdateAllIDReferences() sounds like this pattern could arise quite often?

Reply:
I don't see where this would arise outside of the `operator+=` where it's currently implemented.